ckan/lib/base.py

```diff
@@ -138,6 +138,8 @@ def _allow_caching(cache_force: Optional[bool] = None):
 def _is_valid_session_cookie_data() -> bool:
```

Comment:
I had a hard time understanding why is this function here and why are we by-passing data elements starting with `_` it would be nice to document.

@Zharktas do you remember why is this working like this? I would like to add a docstring to the function.

Reply:
IIRC this logic was to prevent our local directory from filling up with cookie data that is discarded by end users (if all cookie fields are automatically added, don't bother storing them)

Does this change mean that an unauthenticated curl request for a ckan page creates a new file with cookie data on the server? Maybe that's unavoidable for pages with forms now? Is the fix to make sure that old cookie data is cleaned up regularly?

Reply:
The issue with storing too many session files (https://github.com/ckan/ckan/issues/6948) was not directly related to this code but maybe the original reasoning was similar. The bottom line of the session files saga (#6954 first and #7014 later) was that yes, with the CSRF protection enabled there is always a session file (or redis/db record depending on your config) created to store the `_csrf_token`  token (or whatever is called) but is only one per-user / browser session (not per-request as the original issue reported)

The `_is_valid_session_cookie_data()` check was introduced in https://github.com/ckan/ckan/pull/4781 ([here](https://github.com/ckan/ckan/pull/4781/files#diff-68d59b70c2143d203a1aec158f5eecc93c2f6912357ff1f5598baba1067d3595R264)). I guess the reasoning was, "if there is session data added by the user, don't allow caching", and "by added by the user" what was meant was "not starting with underscore", probably because Beaker always adds stuff like `_last_accessed` to the session.

So I don't think this change will affect the current logic to store session files (they will be stored) but we need to think about how it affects caching. Maybe I'm missing something but does having the csrf token in the session (and added as a meta tag) mean that we can't enable caching at all?

There are some [Django docs](https://docs.djangoproject.com/en/4.1/howto/csrf/#using-csrf-protection-with-caching) that might help

Reply:
the issue I mention wasn't created by this PR, so let's investigate it separately